modules/ingester/instance.go

```diff
@@ -187,13 +187,20 @@ func (i *instance) CompleteBlock(blockID uuid.UUID) (uuid.UUID, error) {
 		i.blocksMtx.Unlock()
 		return uuid.Nil, err
 	}
+
+	err = completingBlock.Clear()
```

Comment:
very nice cleanup.  i didn't even recognize that @annanay25 's PR allowed this.  this might also make my wal replay PR viable.

this is inside the lock.  should we do it after?  i don't know the kernel well enough to know why this call might hang, but i guess it might hang?

Reply:
Actually that is a good question.  The Clear() must occur before removing from the list, else it risks the same situation:  file left around unexpectedly (either deleted or not).  

Without making any changes to the overall error handling, it would be safe to move the lock down after the deletion.  The current implementation actually seems to take the lock too early as if an error occurs it immediately unlocks and exits anyways.

Reply:
Actually the delete cannot occur outside of the lock and before the blocklist is updated or else Find could attempt to search block and the file is missing.    It works to do the delete last outside of lock.

Reply:
Nice cleanup in this function!